saged/models.py

```diff
@@ -134,3 +161,348 @@ def predict(self, dataset: np.array) -> np.array:
+    """
+
+    def __init__(self, seed: int = 42) -> None:
+        """
+        The initializer the LogisticRegression class
+
+        seed: The random seed to be used by the model
+        """
+        self.model = sklearn.linear_model.LogisticRegression(random_state=seed)
+
+    def fit(self, dataset: LabeledDataset) -> "LogisticRegression":
+        """
+        Train a model using the given labeled data
+
+        Arguments
+        ---------
+        dataset: The labeled data for use in training
+
+        Returns
+        -------
+        self: The fitted model
+        """
+        X, y = dataset.get_all_data()
+
+        self.model = self.model.fit(X, y)
+        return self
+
+    def predict(self, dataset: UnlabeledDataset) -> np.ndarray:
+        """
+        Use the model to predict the labels for a given unlabeled dataset
+
+        Arguments
+        ---------
+        dataset: The unlabeled data whose labels should be predicted
+
+        Returns
+        -------
+        predictions: A numpy array of predictions
+        """
+        X = dataset.get_all_data()
+        return self.model.predict(X)
+
+    def save_model(self, out_path: str) -> None:
+        """
+        Write the classifier to a file
+
+        Arguments
+        ---------
+        out_path: The path to the file to write the classifier to
+
+        Raises
+        ------
+        FileNotFoundError if out_path isn't openable
+        """
+
+        with open(out_path, 'wb') as out_file:
+            pickle.dump(self, out_file)
+
+    @classmethod
+    def load_model(classobject, model_path):
+        """
+        Read a pickeled model from a file and return it
+
+        Arguments
+        ---------
+        model_path: The location where the model is stored
+
+        Returns
+        -------
+        model: The model saved at `model_path`
+        """
+        with open(model_path, 'rb') as model_file:
+            return pickle.load(model_file)
+
+
+class ThreeLayerClassifier(nn.Module):
+    """ A basic three layer neural net for use in wrappers like FullyConnectedNet """
+    def __init__(self, model_params: dict):
+        super(ThreeLayerClassifier, self).__init__()
+        input_size = model_params['input_size']
+        output_size = model_params['output_size']
+
+        self.fc1 = nn.Linear(input_size, input_size // 2)
+        self.fc2 = nn.Linear(input_size // 2, input_size // 4)
+        self.fc3 = nn.Linear(input_size // 4, output_size)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = F.relu(self.fc1(x))
+        x = F.relu(self.fc2(x))
+        x = self.fc3(x)
+
+        return x
+
+
+class PytorchSupervised(ExpressionModel):
+    """
+    A wrapper class implementing the ExpressionModel API while remaining modular enough
+    to accept any supervised classifier implementing the nn.Module API
+    """
+    def __init__(self,
+                 config: dict,
```

Comment:
Do you know of a good way to document what needs to be in this config, and (if anything) what parameters are optional? I guess as it stands your code will throw a `KeyError` if (for example) the learning rate or seed is missing, but it might be good to have a list of what's necessary somewhere centralized.

Even just documentation pointing to the example config in `test/data/test_config.yml` might serve this purpose.

Reply:
That's a good question. I think it will make sense to write a custom parser function that validates the file, but I'll wait for a bit before doing so to let the config format solidify.

I think for now I'll add a reference in the README and add a TODO to the code